test/controllers/declaration/SealControllerSpec.scala

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.declaration
+
+import base.CSRFUtil._
+import base.CustomExportsBaseSpec
+import base.TestHelper._
+import forms.Choice
+import forms.Choice.choiceId
+import forms.declaration.Seal
+import generators.Generators
+import org.mockito.ArgumentMatchers
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.{reset, verify}
+import org.scalacheck.Arbitrary._
+import org.scalacheck.Gen.listOf
+import org.scalatest.prop.PropertyChecks
+import play.api.data.Form
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import play.twirl.api.Html
+import uk.gov.hmrc.auth.core.InsufficientEnrolments
+import views.html.declaration.seal
+
+class SealControllerSpec extends CustomExportsBaseSpec with Generators with PropertyChecks {
+
+  private val uri = uriWithContextPath("/declaration/add-seal")
+
+  val form: Form[Seal] = Form(Seal.formMapping)
+
+  def view(form: Form[Seal] = form, seals: Seq[Seal])(implicit request: FakeRequest[_]): Html =
```

Comment:
The same propose like in tests for `BorderTransportControllerSpec`

Reply:
updated